File: UMS.Core/Models/Student.cs
using System;
using System.Collections.Generic;

namespace UMS.Core.Models;

public partial class Student
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public string CreatorUserId { get; set; } = null!;

    public DateTime? CreationDate { get; set; }

    public string? ModificationUserId { get; set; }

    public DateTime? ModificationDate { get; set; }

    public bool IsDeleted { get; set; }

    public string? DeletionUserId { get; set; }

    public DateTime? DeletionDate { get; set; }

    public virtual ICollection<StudentSubject> StudentSubjects { get; set; } = new List<StudentSubject>();
}

File: ums.client/src/shared/models/Student.ts
export class Student {
  id: number;
  name: string | null;
  creatorUserId: string;
  creationDate?: Date;
  modificationUserId?: string | null;
  modificationDate?: Date | null;
  isDeleted: boolean;
  deletionUserId?: string | null;
  deletionDate?: Date | null;

  constructor(
    id: number,
    creatorUserId: string,
    isDeleted: boolean,
    name: string | null,
    creationDate?: Date,
    modificationUserId?: string | null,
    modificationDate?: Date | null,
    deletionUserId?: string | null,
    deletionDate?: Date | null
  ) {
    this.id = id;
    this.name = name;
    this.creatorUserId = creatorUserId;
    this.creationDate = creationDate;
    this.modificationUserId = modificationUserId;
    this.modificationDate = modificationDate;
    this.isDeleted = isDeleted;
    this.deletionUserId = deletionUserId;
    this.deletionDate = deletionDate;
  }
}
